moo.js

```diff
@@ -91,16 +91,18 @@
       obj = { match: obj }
     }
 
-    // nb. error implies lineBreaks
+    // nb. error and default imply lineBreaks
     var options = {
       tokenType: name,
-      lineBreaks: !!obj.error,
+      lineBreaks: !!obj.error || !!obj.default,
       pop: false,
       next: null,
       push: null,
       error: false,
+      default: false,
       value: null,
       getType: null,
+      shouldThrow: false,
```

Comment:
`error` is like `fallback` but it consumes until the end of the buffer instead of just to the next valid token. `shouldThrow` throws a syntax error instead of returning the error token.